JCR-3629 - [jcr2spi]RepositoryException lost in o.a.j.jcr2spi.ItemManagerImpl while querying on remote nodes

-- Changed the logic to propagate the RepositoryException 
-- Adapted the logic in LazyItemIterator#skip to handle the checked RepositoryException

git-svn-id: https://svn.apache.org/repos/asf/jackrabbit/trunk@1507190 13f79535-47bb-0310-9956-ffa450edef68

-            while (!itemMgr.itemExists(entry)) {
-                log.debug("Ignoring nonexistent item " + entry);
-                entry = iter.next();
+            boolean itemExists = false;
+            while(!itemExists){
+                try{
+                    itemExists = itemMgr.itemExists(entry);
+                }catch(RepositoryException e){
+                    log.warn("Failed to check that item {} exists",entry,e);
+                }
+                if(!itemExists){
+                    log.debug("Ignoring nonexistent item {}", entry);
+                    entry = iter.next();
+                }
